site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,323 @@
+
+**Response Message Format**
+
+A response to a *remove credentials* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty.
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | No Content, the credentials have been removed successfully. |
+| *404* | Not Found, there are no credentials registered matching the critera given in the request payload. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+# Standard Message Properties
+
+Due to the nature of the request/response message pattern of the operations of the Credentials API, there are some standard properties shared by all of the request and response messages exchanged as part of the operations.
+
+## Standard Request Properties
+
+The following table provides an overview of the properties shared by all request messages regardless of the particular operation being invoked.
+
+| Name             | Mandatory | Location                 | Type           | Description |
+| :--------------- | :-------: | :----------------------- | :------------- | :---------- |
+| *action*         | yes       | *application-properties* | UTF-8 *string* | MUST be set to the value defined by the particular operation being invoked. |
+| *correlation-id* | no        | *properties*             | *message-id*   | MAY contain an ID used to correlate a response message to the original request. If set, it is used as the *correlation-id* property in the response, otherwise the value of the *message-id* property is used. |
+| *message-id*     | yes       | *properties*             | UTF-8 *string* | MUST contain an identifier that uniquely identifies the message at the sender side. |
+| *reply-to*       | yes       | *properties*             | UTF-8 *string*  | MUST contain the source address that the client wants to received response messages from. This address MUST be the same as the source address used for establishing the client's receive link (see [Preconditions]({{< relref "#preconditions" >}}). |
+
+## Standard Response Properties
+
+The following table provides an overview of the properties shared by all response messages regardless of the particular operation being invoked.
+
+| Name             | Mandatory | Location                 | Type            | Description |
+| :--------------- | :-------: | :----------------------- | :-------------- | :---------- |
+| *correlation-id* | yes       | *properties*             | *message-id*    | Contains the *message-id* (or the *correlation-id*, if specified) of the request message that this message is the response to. |
+| *device_id*      | yes       | *application-properties* | UTF-8 *string*  | Contains the ID of the device. |
```

Comment:
Yes, I thought about that as well. However, I also recall a discussion about the Registration API we had last year during our f2f workshop in Berlin about the general question of whether it makes sense to include e.g. tenant and device ID as properties on all messages or not. IIRC Gordon proposed to also have these properties even if e.g. the address already contains this information as well so that e.g. the Dispatch Router could more easily access this information and use it for routing decisions etc.
So, I do agree that this is redundant information but I also see Gordon's point, so at the moment, I would actually go for including it in the JSON (adding tenant as well) while keeping the properties as well. WDYT?